src/CountryHandler/Slovakia.php

```diff
@@ -24,14 +24,28 @@ final class Slovakia extends CountryHandler
      */
     public const LENGTH = 10;
 
+    public const PATTERN = '([1-9]\\d[234789]\\d{7})|(\\d{2}[0156]\\d[0-3]\\d{4,5})';
+
     protected function hasValidLength(string $tin): bool
     {
         $c1c2 = substr($tin, 0, 2);
 
         if (54 > $c1c2) {
-            return $this->matchLength($tin, self::LENGTH - 1);
+            return $this->matchLength($tin, self::LENGTH) || $this->matchLength($tin, self::LENGTH - 1);
         }
 
         return parent::hasValidLength($tin);
     }
+
+    public function hasValidRule(string $tin): bool
+    {
+        if (strlen($tin) === 10
+            && (int) $tin % 11 === 0) {
+            return true;
+        }
+        if (strlen($tin) === 10) {
+            return ((int) substr($tin, 0, 9) % 11) % 10 === (int) substr($tin, 9, 1);
+        }
```

Comment:
```suggestion
        if (strlen($tin) === 10) {
          if ( (int) $tin % 11 === 0) {
            return true;
          }
          
          return ((int) substr($tin, 0, 9) % 11) % 10 === (int) substr($tin, 9, 1);        
        }
```

Reply:
On the EU checker page (https://ec.europa.eu/taxation_customs/tin/#/check-tin) you can try e.g. 2822030541
This TIN is valid, even if it has 10 digits and starts with < 54.

We decompiled the Java lib you can find here:
https://taxation-customs.ec.europa.eu/document/download/07664928-b4f3-47a5-81f7-438767dc01cb_en?v=1721203258495

The modulo 11 check has those 2 modes in the decompiled lib.